### PR TITLE
ISPN-13842 Infinispan indexing annotations

### PR DESCRIPTION
**Pull request by fax4ever**
https://issues.redhat.com/browse/ISPN-13842

### CODE REVIEW

**Comment by fax4ever** [comment]
I'm going to rebase on top of the current master changes.

**Comment by karesti** [comment]
@fax4ever needs rebase

**Comment by fax4ever** [comment]
> @fax4ever needs rebase

Hi Katia. I'm doing it.

**Comment by tristantarrant** [comment]
Be careful with adding the `infinispan-api` dependency to other modules: it requires JDK 11 (most of our stuff is JDK 8 compatible). I'd make it `optional`

**Comment by fax4ever** [comment]
> Be careful with adding the `infinispan-api` dependency to other modules: it requires JDK 11 (most of our stuff is JDK 8 compatible). I'd make it `optional`

I rebased, made some changes, but I still need to address well the optionality of the `infinispan-api` dependency.

File: documentation/src/main/asciidoc/topics/code_examples/AuthorEmbedded.java [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
 
 public class Author {
-   @FullTextField
+   @Text
    String name;
 
-   @FullTextField
+   @Text
    String surname;
 }
```
Comment (tristantarrant) [inline documentation/src/main/asciidoc/topics/code_examples/AuthorEmbedded.java:11]: missing line feed

**Comment by fax4ever** [comment]
I've done with the changes. Let me know if we can improve this contribution further.

**Comment by tristantarrant** [comment]
Merged, thanks

**Comment by fax4ever** [comment]
Thanks!

**Pull request merged**